src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -501,8 +555,20 @@ private ButtonElement buildButton(String text, String value, ActionId actionId,
         return button;
     }
 
-    private LayoutBlock buildPlainTextBlock(String text) {
-        return SectionBlock.builder().text(PlainTextObject.builder().text(text).build()).build();
+    private LayoutBlock buildPlainTextBlock(String text, String id) {
+        return SectionBlock.builder().text(PlainTextObject.builder().text(text).build()).blockId(id).build();
+    }
+
+    private Optional<LayoutBlock> getSectionBlockWithId(List<LayoutBlock> blocks, String id) {
+        for (LayoutBlock block : blocks) {
+            if(block.getClass().getName().equals("com.slack.api.model.block.SectionBlock")) {
```

Comment:
What else could it be?

Reply:
There are also dividerblocks that cannot be parsed into sectionblocks, and regular layout blocks don't have block id's so they must be parsed.